allauth/socialaccount/providers/dataporten/provider.py

```diff
@@ -0,0 +1,96 @@
+from allauth.socialaccount.providers.base import ProviderAccount
+from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
+
+
+class DataportenAccount(ProviderAccount):
+    def get_avatar_url(self):
+        '''
+        Returns a valid URL to an 128x128 .png photo of the user
+        '''
+        # Documentation for user profile photos can be found here:
+        # https://docs.dataporten.no/docs/oauth-authentication/
+        base_url = 'https://api.dataporten.no/userinfo/v1/user/media/'
+        return base_url + self.account.extra_data['profilephoto']
+
+    def to_str(self):
+        '''
+        Returns string representation of a social account. Includes the name
+        of the user.
+        '''
+        dflt = super(DataportenAccount, self).to_str()
+        return '%s (%s)' % (
+            self.account.extra_data.get('name', ''),
+            dflt,
+        )
+
+
+class DataportenProvider(OAuth2Provider):
+    id = 'dataporten'
+    name = 'Dataporten'
+    account_class = DataportenAccount
+
+    def extract_uid(self, data):
+        '''
+        Returns the primary user identifier, an UUID string
+        See: https://docs.dataporten.no/docs/userid/
+        '''
+        return data['userid']
+
+    def extract_extra_data(self, data):
```

Comment:
I can remove this unecessary function definition if desired. I added it as it felt more explicit at the time.